graylog2-web-interface/src/components/cluster/GraylogClusterOverview.tsx

```diff
@@ -0,0 +1,187 @@
+  const { traffic } = useStore(ClusterTrafficStore);
+  const [graphWidth, setGraphWidth] = useState(600);
+  const eventThrottler = useRef(new EventHandlersThrottler());
+  const containerRef = useRef(null);
+  const licensePlugin = PluginStore.exports('license');
+
+  useEffect(() => {
+    const _resizeGraphs = () => {
+      const { clientWidth } = containerRef.current;
+
+      setGraphWidth(clientWidth);
+    };
+
+    const _onResize = () => {
+      eventThrottler.current.throttle(() => _resizeGraphs());
+    };
+
+    ClusterTrafficActions.getTraffic();
+    window.addEventListener('resize', _onResize);
+
+    if (containerRef) {
+      _resizeGraphs();
+    }
+
+    return () => {
+      window.removeEventListener('resize', _onResize);
+    };
+  }, [containerRef]);
+
+  const renderClusterInfo = () => {
+    let content = <Spinner />;
+
+    if (nodes) {
+      const { clusterId, nodeCount } = nodes;
+
+      content = (
+        <StyledDl className="system-dl">
+          <dt>Cluster ID:</dt>
+          <dd>{clusterId || 'Not available'}</dd>
+          <dt>Number of nodes:</dt>
+          <dd>{nodeCount}</dd>
+        </StyledDl>
+      );
+    }
+
+    return content;
+  };
+
+  const renderTrafficGraph = () => {
+    const TrafficGraphComponent = licensePlugin[0]?.EnterpriseTrafficGraph || TrafficGraph;
+    let sumOutput = null;
+    let trafficGraph = <Spinner />;
+
+    if (traffic) {
+      const bytesOut = _.reduce(traffic.output, (result, value) => result + value);
+
+      sumOutput = <small>Last 30 days: {NumberUtils.formatBytes(bytesOut)}</small>;
+
+      trafficGraph = (
+        <TrafficGraphComponent traffic={traffic.output}
+                               from={traffic.from}
+                               to={traffic.to}
+                               width={graphWidth} />
+      );
+    }
+
+    return (
+      <>
+        <StyledH3 ref={containerRef}>Outgoing traffic {sumOutput}</StyledH3>
+        {trafficGraph}
+      </>
+    );
+  };
+
+  const renderHeader = () => {
+    return <StyledH2>Graylog cluster</StyledH2>;
+  };
+
+  const renderDefaultLayout = () => {
+    return (
+      <Row className="content">
+        <Col md={12}>
+          {renderHeader()}
+          {renderClusterInfo()}
+          <hr />
+          {children}
+          <Row>
+            <Col md={12}>
+              {renderTrafficGraph()}
+            </Col>
+          </Row>
+        </Col>
+      </Row>
+    );
+  };
+
+  const renderCompactLayout = () => {
+    return (
+      <Row className="content">
+        <Col md={12}>
+          {renderHeader()}
+          <Row>
+            <Col md={6}>
+              {renderClusterInfo()}
+              <hr />
+              {children}
+            </Col>
+            <Col md={6}>
+              {renderTrafficGraph()}
+            </Col>
+          </Row>
+        </Col>
+      </Row>
+    );
+  };
+
+  if (layout === 'compact') {
+    return renderCompactLayout();
```

Comment:
I think we should try moving away of using functions from this purpose and use inner components instead. At least for me it is clearer what we do that way.